gemonlinedb/include/gem/onlinedb/XMLSerializationData.h

```diff
@@ -0,0 +1,248 @@
+#include "gem/onlinedb/DataSet.h"
+#include "gem/onlinedb/PartReference.h"
+#include "gem/onlinedb/Run.h"
+#include "gem/onlinedb/detail/RegisterData.h"
+
+namespace gem {
+    namespace onlinedb {
+
+        /**
+         * @brief Convenience alias.
+         */
+        using DOMDocumentPtr = std::unique_ptr<xercesc::DOMDocument>;
+
+        /**
+         * @brief Represents the contents of XML files.
+         *
+         * This class is used to represent the contents of XML files as C++
+         * objects before they are serialized and after they are read. It is
+         * used as an indermediate step when (un)serializing configuration data.
+         *
+         * @tparam ConfigurationTypeT The configuration class serialized in the
+         *                            XML file.
+         */
+        template<class ConfigurationTypeT>
+        class XMLSerializationData
+        {
+        public:
+            /**
+             * @brief Convenience typedef.
+             */
+            using ConfigurationType = ConfigurationTypeT;
+
+        private:
+            Run run;
+            std::vector<DataSet<ConfigurationType>> dataSets;
+
+        public:
+            /**
+             * @brief Returns the run information from the file.
+             */
+            Run getRun() const { return run; }
+
+            /**
+             * @brief Sets the run for which to generate XML data.
+             */
+            void setRun(const Run &run) { this->run = run; }
+
+            /**
+             * @brief Gets the list of parsed datasets.
+             */
+            std::vector<DataSet<ConfigurationType>> getDataSets() const
+            {
+                return dataSets;
+            }
+
+            /**
+             * @brief Adds a dataset to be serialized
+             */
+            void addDataSet(const DataSet<ConfigurationType> &dataSet)
+            {
+                dataSets.push_back(dataSet);
+            }
+
+            /**
+             * @brief Reads data from a DOM document.
+             *
+             * @warning No validation is performed as part of this function.
+             *          Feeding a document that doesn't conform to the schema
+             *          results in undefined behaviour.
+             */
+            void readDOM(const DOMDocumentPtr &dom);
+
+            /**
+             * @brief Generates a DOM document representing data previously set
+             *        using @ref setRun and @ref addDataSet.
+             */
+            DOMDocumentPtr makeDOM() const;
+        };
+
+        // Utility functions for readDOM
+        namespace detail {
+
+            /**
+             * @brief Extracts run information from XML data.
+             */
+            Run getRun(const DOMDocumentPtr &dom);
+
+            /**
+             * @brief Queries the list of data sets present in @c dom
+             */
+            xercesc::DOMXPathResult *queryDataSets(const DOMDocumentPtr &dom);
```

Comment:
Did you look into using `shared_ptr` or `unique_ptr` to encapsulate the `xercesc` pointer objects?
I know it's not always trivial, but I did a bunch of this with the `xdaq` core pointer objects and it helped clean up a few memory leaks.